pytorch_lightning/callbacks/gradient_accumulation_scheduler.py

```diff
@@ -74,9 +74,16 @@ def __init__(self, scheduling: Dict[int, int]):
     def going_to_accumulate_grad_batches(self):
         return any(v > 1 for v in self.scheduling.values())
 
-    def on_train_epoch_start(self, trainer, pl_module):
-        epoch = trainer.current_epoch
+    def _get_current_accumulate_grad_batches(self, epoch: int) -> int:
```

Comment:
why do we need to split this? can't we just inline this to `on_train_epoch_start`? Same for `compute_accumulate_grad_batches. IMO this should all be inlined since this seems to be the only place this is called.

Reply:
Removed the extra function.

Reply:
This is required as we need to compute the current `accumulate_grad_batches` and calling `on_train_epoch_start` there could be confusing for readers.